internal/provider/deployment.go

```diff
@@ -114,7 +114,7 @@ func (d *DeploymentDeployer) assignNodesIPs() error {
 		return errors.Wrapf(err, "invalid ip %s", d.IPRange)
 	}
 	for _, vm := range d.VMs {
-		if vm.IP != "" && cidr.Contains(net.ParseIP(vm.IP)) && !isInByte(usedIPs, net.ParseIP(vm.IP)[3]) {
+		if vm.IP != "" && cidr.Contains(net.ParseIP(vm.IP)) && !includes(usedIPs, net.ParseIP(vm.IP)[3]) {
```

Comment:
split the condition  `vm.IP == "" { continue } `
and please add a comment to clarify what is `[3]` is used for